backend/workflow_manager/workflow_v2/views.py

```diff
@@ -108,6 +126,53 @@ def perform_create(self, serializer: WorkflowSerializer) -> Workflow:
             raise WorkflowGenerationError
         return workflow
 
+    def partial_update(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        """Override partial_update to handle sharing notifications."""
+        # Get the workflow instance before update
+        workflow = self.get_object()
+
+        # Store current shared users for comparison
+        current_shared_users = set(workflow.shared_users.all())
+
+        # Perform the standard partial update
+        response = super().partial_update(request, *args, **kwargs)
+
+        # If update was successful and shared_users field was modified
+        if (
+            response.status_code == 200
+            and "shared_users" in request.data
+            and NOTIFICATION_PLUGIN_AVAILABLE
+        ):
+            try:
+                # Get updated workflow to compare shared users
+                workflow.refresh_from_db()
+                new_shared_users = set(workflow.shared_users.all())
+
+                # Find newly added users
+                newly_shared_users = new_shared_users - current_shared_users
+
+                if newly_shared_users:
+                    # Send notification to newly shared users
+                    sharing_notification_service.send_sharing_notification(
+                        resource_type=ResourceType.WORKFLOW.value,
+                        resource_name=workflow.workflow_name,
+                        resource_id=str(workflow.id),
+                        shared_by=request.user,
+                        shared_to=list(newly_shared_users),
+                        resource_instance=workflow,
+                    )
+
+                    logger.info(
+                        f"Sent sharing notifications for workflow {workflow.id} "
+                        f"to {len(newly_shared_users)} users"
+                    )
+
+            except Exception as e:
+                # Log error but don't fail the update operation
+                logger.error(f"Failed to send sharing notification: {str(e)}")
```

Comment:
```suggestion
                logger.exception(f"Failed to send sharing notification, continuing update though: {str(e)}")
```
Fix any line length related issue